apps/next-evans/components/pages/home/news-feed/news-feed.ts

```diff
@@ -0,0 +1,507 @@
+import { t } from '@friends-library/locale';
+import { htmlShortTitle } from '@friends-library/adoc-utils';
+import type { Lang } from '@friends-library/types';
+import type { Audiobook, DocumentWithMeta, NewsFeedType } from '@/lib/types';
+import { months } from '@/lib/dates';
+import { LANG } from '@/lib/env';
+import { isNotNull } from '@/lib/utils';
+import { getDocumentUrl } from '@/lib/friend';
+
+export interface FeedItem {
+  month: string;
+  day: string;
+  year: string;
+  type: NewsFeedType;
+  title: string;
+  description: string;
+  url: string;
+  date: string;
+}
+
+type NewsfeedDocumentProps = Pick<
+  DocumentWithMeta,
+  | 'authorName'
+  | 'dateAdded'
+  | 'id'
+  | 'title'
+  | 'authorSlug'
+  | 'slug'
+  | 'editions'
+  | 'mostModernEdition'
+  | 'altLanguageId'
+>;
+
+export async function getNewsFeedItems(
+  lang: Lang,
+  documents: {
+    en: Array<NewsfeedDocumentProps>;
+    es: Array<NewsfeedDocumentProps>;
+  },
+): Promise<FeedItem[]> {
+  const outOfBandEvents = getOutOfBandEvents();
+  const recentAdditions = getRecentAdditions(`es`, documents.es, documents.en).concat(
+    LANG === `en` ? getRecentAdditions(`en`, documents.en) : [],
+  );
+  const recentAudios = getRecentAudios(documents[lang]);
+  return recentAdditions
+    .concat(outOfBandEvents.filter((item) => item.lang.includes(lang)))
+    .concat(recentAudios)
+    .sort((a, b) => new Date(b.date).getTime() - new Date(a.date).getTime())
+    .slice(0, MAX_NUM_NEWS_FEED_ITEMS);
+}
+
+function getRecentAdditions(
+  lang: Lang,
+  documents: Array<NewsfeedDocumentProps>,
+  altLanguageDocuments: Array<NewsfeedDocumentProps> = [],
+): FeedItem[] {
+  const type = LANG === `en` && lang === `es` ? `spanish_translation` : `book`;
+  return documents
+    .sort((a, b) => new Date(b.dateAdded).getTime() - new Date(a.dateAdded).getTime())
+    .map((doc) => {
+      return {
+        type,
+        title:
+          lang === `en`
+            ? htmlShortTitle(doc.title)
+            : `${htmlShortTitle(doc.title)}${LANG === `en` ? ` &mdash; (Spanish)` : ``}`,
+        ...dateFields(doc.mostModernEdition.impressionCreatedAt, LANG),
+        description:
+          type === `book`
+            ? LANG === `en`
+              ? `Download free eBook or pdf, or purchase a paperback at cost.`
+              : `Descárgalo en formato ebook o pdf, o compra el libro impreso a precio de costo.`
+            : `${doc.authorName}'s <em>${
+                altLanguageDocuments.find((altDoc) => altDoc.altLanguageId === doc.id)
+                  ?.title ?? ``
+              }</em> is now translated and available on the Spanish site.`,
+        url:
+          type === `spanish_translation`
+            ? `https://bibliotecadelosamigos.org/${doc.authorSlug}/${doc.slug}`
+            : `/${doc.authorSlug}/${doc.slug}`,
```

Comment:
i have a two-part proposal:

first, i think this line and the one above it should be calling out to our `getDocumentUrl()` helper function. not so much because it's shorter, as because it preserves a single source of truth. if we ever wanted to change how our urls are structured, it would be nice to know we could change that one function, and not miss any hand-constructed urls sprinkled thru the code base.

second, is i think it might be nice to overload that function so that it could take a `{authorSlug: string; slug: string }` arg as well as the two string args. then, it would be more convenient to use. if you had two strings, pass em; if you have a document-like object, pass it.